app/src/components/user_details_form.rs

```diff
@@ -402,6 +406,7 @@ impl UserDetailsForm {
             self.user.avatar = Some(avatar);
         }
         self.just_updated = true;
+        self.avatar_event_bus.send(Request::Update);
```

Comment:
Why not send the avatar directly in the bus? And we check whether the avatar has changed (before setting `self.user.avatar`).